middleware/auth.js

```diff
@@ -0,0 +1,23 @@
+const jwt = require('jsonwebtoken');
+const User = require('../models/user');
+
+module.exports = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization.split(' ')[1]; 
+    if (!token) {
+      return res.status(401).json({ message: "Not authorized" });
+    }
+
+    const decoded = jwt.verify(token, process.env.JWT_SECRET);
+    const user = await User.findById(decoded.userId);
```

Comment:
Sprawdź, czy użytkownik ma aktywną sesję. Oznacza to, że należy sprawdzić, czy token przechowywany w bazie danych jest taki sam jak token przesyłany w żądaniu. Dzięki temu, nawet jeśli czas ważności tokena JWT nie minął, będzie aktywny tylko jeden token.

Reply:
Poprawiłem.